lab.py

```diff
@@ -0,0 +1,8 @@
+def getlist(dictionary):
+    global list_format
+    base_list = []
+    for i in dictionary.keys():
+        base_list.append(i)
+        list_format = ", ".join(base_list)
+
+    return list_format
```

Comment:
A few things here:

* You should never use verbs like `get`, `set`, `create`, etc. These are very C like and do not convey any useful information in Python. It just adds more characters to a parameter or function.
* Try to avoid at all cost `global`. Also in this case it was not necessary to define a global variable. When you have to use them, it usually means that the architecture is not right. There are just a few cases where globals are really needed.
* In your algorithm, you are rewriting `list_format` at every iteration with the last state `base_list`. So you can actually just do the join at the end.
* This brings me to the fact that `join` alone would do what you wanted as it takes an iterable. Your code is equivalent to this: `", ".join(dictionary)`.

Reply:
* got it :)
* global was mostly pycharm asking me to declare the variable, I'll have more question about this when we grab a bite
* ok for the join at the end I'll try something
* however I used the loop to pick only the keys on list since i want them to be listed and not the value ( assuming the join would take the keys AND values from the dict.
Thanks for reviewing !

Reply:
When you iterate on a dict, you just get the keys and not the values. So this is already doing what you want 😉
Try `", ".join(dictionary)`, you will see.

Reply:
When you iterate on a dict, you just get the keys and not the values. So this is already doing what you want 😉
Try ", ".join(dictionary), you will see.

Reply:
Hot damn 🔥